cns/api.go

```diff
@@ -175,6 +176,13 @@ type NodeConfiguration struct {
 type IPAMPoolMonitor interface {
 	Start(ctx context.Context, poolMonitorRefreshMilliseconds int) error
 	Update(scalar nnc.Scaler, spec nnc.NodeNetworkConfigSpec) error
+	IPAMPoolMonitorFreeIps() IPAMPoolMonitorFreeIps
```

Comment:
I think there will be more state that can be surfaced through here. I think to convey proper intent, this could be renamed to `GetStateSnapshot()` and the struct below to `IpamPoolMonitorStateSnapshot`. This way, more data can be added without making the intention of the struct ambiguous.

Reply:
makes sense, I was confused as to what to name so this was temp, but the StateSnapshot seems right to convey that. Have refactored.